configure.ac

```diff
@@ -572,6 +572,17 @@ AC_ARG_ENABLE([thread-sanitizer],
   AS_HELP_STRING([--enable-thread-sanitizer], [enable ThreadSanitizer support for detecting data races]))
 AC_ARG_ENABLE([memory-sanitizer],
   AS_HELP_STRING([--enable-memory-sanitizer], [enable MemorySanitizer support for detecting uninitialized memory reads]))
+AC_ARG_WITH([openssl],
```

Comment:
I would prefer we call this `--with-crypto=<internal|openssl|...>` and I would rather do so from the start so we don't change it several times over.

For context, I'll probably be adding a `--with-crypto=nettle` option (nettle is gnutls' crypto library).  I would like to point out here that since OpenSSL is licensed under the Apache 2.0 license, FRR builds against OpenSSL can only be distributed by bumping to GPLv3 since GPLv2 is not Apache 2.0 compatible.

(Nettle is licensed under "GPLv2+ or LGPLv3+", so the GPLv2+ choice there is a better match for FRR.)

Reply:
FRR is licensed under GPLv2+ am I right? This should make it compatible with ASL-2.0. Also, in RHEL and in Fedora we have "operating system exception" for later version of openssl which mitigates this issue. 

Does this mean that you don't want to use openssl at all? Or just that you would like to have more options here? I can amend the PR to use gnutls as well or to use gnutls exclusively.

Reply:
@pogojotz I was wondering about the HAVE_LIBCRYPT in lib/zebra.h. The -lcrypt is added as part of $PYTHON_LIBS during build, but I could not find any usage of it and are the crypt and crypto even related?

Reply:
`-lcrypt` has no relation to crypto/`-lcrypto`

@mruprich FRR policy is that for everything that is not a plugin / optional extension, it must work with GPLv2 and GPLv3, i.e. pushing to GPLv3 only is not accepted. That means ASL-2.0 won't do since it requires this bump.

However, the code here is optional because we have the switch that chooses between internal crypto and OpenSSL, so anyone who wants to use GPLv2 can do so by using the internal crypto.

Nettle as a choice here could replace our internal crypto options so we can remove them, since it is available under GPLv2.  Either way, I think this is a situation where multiple choices are acceptable to have.

Reply:
Policy is written down here btw:
https://github.com/FRRouting/frr/blob/master/doc/developer/workflow.rst#L227-L235

It's very confusingly worded, sorry about that, the person who wrote it is not a native English speaker :).  The person was more concerned about using GPLv3, but the implication was that it must be possible to use FRR under both GPLv2 or GPLv3 at the liberty of the person who packages it.